packages/merge-styles/src/styleToClassName.ts

```diff
@@ -42,7 +42,7 @@ function extractRules(
       const expandedRules = stylesheet.argsFromClassName(arg);
 
       if (expandedRules) {
-        extractRules(expandedRules, rules);
+        extractRules(expandedRules, rules, currentSelector);
```

Comment:
The tests are easy to write. Run `npm run start-test` in the merge-styles package folder and then just add tests. I recommend when hitting these cases, make a test first to validate it fails your assertion, then fix the bug.

Reply:
Added tests